ssz-rs-derive/src/lib.rs

```diff
@@ -369,12 +380,22 @@ fn derive_merkleization_impl(data: &Data) -> TokenStream {
                 Some(field_name) => quote_spanned! { f.span() =>
                     let chunk = self.#field_name.hash_tree_root()?;
                     let range = #i*#BYTES_PER_CHUNK..(#i+1)*#BYTES_PER_CHUNK;
-                    chunks[range].copy_from_slice(chunk.as_ref());
+                    let chunks_byte_count = chunks.len();
+                    // index is safe as long as Node::len() == BYTES_PER_CHUNK
+                    chunks
+                        .get_mut(range)
+                        .ok_or(MerkleizationError::InputInsufficient(chunks_byte_count))?
+                        .copy_from_slice(chunk.as_ref());
                 },
                 None => quote_spanned! { f.span() =>
                     let chunk = self.0.hash_tree_root()?;
                     let range = #i*#BYTES_PER_CHUNK..(#i+1)*#BYTES_PER_CHUNK;
-                    chunks[range].copy_from_slice(chunk.as_ref());
+                    let chunks_byte_count = chunks.len();
+                    // index is safe as long as Node::len() == BYTES_PER_CHUNK
+                    chunks
+                        .get_mut(range)
+                        .ok_or(MerkleizationError::InputInsufficient(chunks_byte_count))?
+                        .copy_from_slice(chunk.as_ref());
```

Comment:
somewhat strong preference for leaving these alone

very open to either a comment `// SAFETY: ...` or some kind of `debug_assert!` that the invariant holds

but this was the intention (impl is auto-generated and can just directly read/write the buffer)